crates/legacy/src/builder_state.rs

```diff
@@ -661,6 +662,13 @@ impl<Types: NodeType> BuilderState<Types> {
 
         for tx in txn_commitments.iter() {
             self.txns_in_queue.remove(tx);
+            // set the status of tx as sequenced, this doesn't mean tx is already decided, but it means tx is included in a builder state, it will either be sequenced or skipped
+            let _ = self.global_state.write_arc().await.set_tx_status(
+                *tx,
+                TransactionStatus::Sequenced {
+                    block: self.parent_block_references.view_number.u64(),
```

Comment:
view number != block height
block (leaf) height is `leaf.height()` here

Reply:
Ooooh, I really haven't thought through where to mark it as `sequenced` yet, now I'm going to rethink about it.